### PR TITLE
Fix singularity local launcher

### PR DESCRIPTION
**Pull request by ashao**
When not using Apptainer, the singularity tests were failing on
the local launcher because the SmartSim-specified run directories
were not being mounted. To solve this, when the launch command
is constructed, the working directory is passed to the container's
command constructor and automatically bound. For advanced users,
the working directory can be specified when the container object
is created in which case that directory takes precedence.

### CODE REVIEW

File: tests/test_containers.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def test_singularity_basic(fileutils):
 
     script = fileutils.get_test_conf_path("sleep.py")
     model.attach_generator_files(to_copy=[script])
-    exp.generate(model)
+    exp.generate(model, overwrite=True)
```
Comment (al-rigazzi) [inline tests/test_containers.py:66]: Why is `overwrite=True` needed? Theoretically, test directories are wiped out before test execution (and after, if all tests passed). The experiment tree should not be there.

**Review by al-rigazzi — APPROVED** [review]
Good stuff, Andrew! I'm so glad you identified and fixed this issue! I just added a comment, if you could address that, then it LGTM

File: tests/test_containers.py (hunk 1 above)
Comment (ashao) [reply tests/test_containers.py:66]: While running the tests manually (i.e. `pytest test_containers.py`), the output didn't seem to be consistently cleared. Happy to take this out if this was some weird edge case and/or detrimental to debugging test failures.

Comment (ashao) [reply tests/test_containers.py]: Yep, you were totally right...I must have just phase shifted into a parallel universe briefly.
